### PR TITLE
feat: Add Transaction class with verification logic

### LINKED ISSUE #8: Create a transaction Class for request to the rest api for publishers
Add Initial verify Method to Class

**Description:**

We need to implement a verify method in the class to validate transaction data and ensure alignment with the backend settings. This method will handle the following properties:

**Properties:**

1. Domain: The domain of the publisher to match with the backend settings.
2. Article URL: The URL of the article being accessed (currently blocked).
3. Publisher Name: The name of the publisher, as configured in the publisher's backend.
4. External UUID: A unique identifier (UUID) for the article.
5. Magazine Key/Token: A unique key or token for identifying the magazine.
6. Checkout Token: A token from the transaction to verify its authenticity.

Acceptance Criteria:

- The verify method should validate these properties against the backend.
-  The implementation should handle invalid or missing values gracefully, providing clear error messages where applicable.
-  Unit tests should be included to cover all possible scenarios.

Notes:

- Ensure the method follows existing coding conventions and integrates seamlessly with the class.
- Discuss edge cases, such as mismatched domains or expired checkout tokens.

### PR DESCRIPTION
**Pull request by damasch**
- Introduced the `Transaction` class to handle transaction data.
- Implemented properties for domain, URL, name, external article ID, magazine key, and checkout token.
- Added a constructor to initialize transaction properties.
- Included a magic getter method for dynamic property access.
- Developed a `verify` method to validate transactions against the backend API.
- Integrated `Config` and `NewsLadderRequest` for API interaction.

This implementation establishes the foundation for managing and verifying transactions in the NewsLadder Publisher API.

closes #8

### CODE REVIEW

**Comment by damasch** [comment]
Unit test are comming

File: src/Transaction.php [hunk 1]
```diff
@@ -2,5 +2,98 @@
 
 namespace NewsLadder\PublisherAPI;
 
+use NewsLadder\PublisherAPI\Config;
+use NewsLadder\PublisherAPI\Request;
+
```
Comment (pascaliske) [inline src/Transaction.php:7]: Within the sub-namespace the use statement can be skipped and the class can be used directly, e.g. `Config` in this case.

```suggestion
```

File: src/Transaction.php [hunk 2]
```diff
@@ -2,5 +2,98 @@
...
+        $payload = [
+            'domain' => $this->domain,
+            'url' => $this->url,
+            'name' => $this->name,
+            'articleEID' => $this->external_article_id,
+            'magazineKey' => $this->magazine_key,
+            'checkoutToken' => $this->checkout_token,
+        ];
+        $config = new Config();
+        $url = $config.get("origins", "api_url") . '/transaction/verify';
```
Comment (pascaliske) [inline src/Transaction.php:93]: JS is calling. This should be converted to PHP syntax. Also using `sprintf` makes the code more readable in my opinion.

```suggestion
        $url = sprintf("%s/transaction/verify", $config->get("origins", "api_url"));
```

**Review by pascaliske — CHANGES_REQUESTED** [review]

**Review by pascaliske — APPROVED** [review]
